tests/components/cookidoo/test_config_flow.py

```diff
@@ -258,3 +258,25 @@ async def test_flow_reauth_error_and_recover(
     assert result["reason"] == "reauth_successful"
 
     assert len(hass.config_entries.async_entries()) == 1
+
+
+async def test_flow_reauth_init_data_already_configured(
+    hass: HomeAssistant,
+    mock_cookidoo_client: AsyncMock,
+    cookidoo_config_entry: MockConfigEntry,
+) -> None:
+    """Test we abort user data set when entry is already configured."""
+
+    cookidoo_config_entry.add_to_hass(hass)
+
+    result = await cookidoo_config_entry.start_reauth_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_EMAIL: EMAIL, CONF_PASSWORD: "new-password"},
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
```

Comment:
You seem to be aborting if entry is trying to change only the password.
What you want is to ignore the "current" entry and only abort if the "current" entry is trying to use the same email as "second" entry.

Reply:
@epenet I dont understand, it is quite normal to just change the password, in that case you have the same email again, no?

Reply:
Ahh, or we have to check if it is the same, which is OK, but if new email, then we abort when there is already a matching config entry.
```
if user_input[CONF_EMAIL] != reauth_entry[CONF_EMAIL]:
    self._async_abort_entries_match(
        {CONF_EMAIL: user_input[CONF_EMAIL]}
    )
```

Reply:
Yes - exactly. Your check and your test is wrong.

You only need to use `self._async_abort_entries_match` if "reauth_entry email" is different from "user_input email"
```python
    if reauth_entry_email != user_input_email:
        self._async_abort_entries_match({email: user_input_email})
```

Reply:
ok, got it.

Reply:
@epenet I think this should be the test needed